accepted/0000-salt-new-maintaining-workflow.md

```diff
@@ -0,0 +1,252 @@
+- Start Date: 2024-11-14
+
+# Summary
+[summary]: #summary
+
+This RFC proposed an improved workflow for maintaining the Salt package for openSUSE/SUSE distributions, and therefore for Uyuni and SUSE Manager.
+
+# Motivation
+[motivation]: #motivation
+
+Our current workflow for maintaining Salt requires manual user intervention after the changes are merged into our `openSUSE/salt` codebase, in order to make this changes available in OBS. Moreover, the Salt spec file and patches are tracked in a separated GitHub repository `openSUSE/salt-packaging`, that is also used to generate the changelog entries for the final RPM.
+
+All these steps needs to be performed manually, with the help of some tooling, to eventually create a manual Submit Request to our Salt package in OBS.
+
+With Salt Extensions appearing now in the upcoming Salt 3008 release, we want to introduce a new workflow that suits better and solves some the limitations we currently have. 
+
+The purpose of this RFC is:
+- Define an new workflow for Salt that reduces user intervention to zero after a given PR is merged in `openSUSE/salt` repository until getting the package ready to consume at OBS.
+- Make the Salt maintaining workflow aligned with openSUSE practices.
+- Provide a workflow that can also work the same way with packaged Salt Extensions.
+- Deprecate the usage of `salt-packaging` repository.
+ 
+# Detailed design
+[design]: #detailed-design
+
+In this new workflow the `openSUSE/salt` GitHub repository will become the unique source of thrust, and it will contain:
+
+- Salt codebase
+- Packaging artifacts: spec file, changelog and extra sources
+- OBS workflow file
+
+Taking inspiration from the [OBS/SCM integration guide](https://openbuildservice.org/help/manuals/obs-user-guide/cha-obs-scm-ci-workflow-integration), the new workflow will use OBS workflows and GitHub Webhooks to automate pulling the changes from GitHub to OBS.
+
+In addition to this, as new Jenkins job will take care of making the OBS package ready to be submitted to openSUSE or SLE.
+
+This is how the proposed OBS structure would look like:
+
+- `systemsmanagement:saltstack/salt`:
+   * no services enabled - package ready to be submitted to openSUSE or SLE.
+   * branched from `systemsmanagement:saltstack:github/salt`
+- `systemsmanagement:saltstack:github/salt`:
+   * services enabled
+   * package building based on `openSUSE/release/xxxx` GitHub branch.
+- `systemsmanagement:saltstack:github:CI:...:PR-XXXX/salt`:
+   * services enabled
+   * package building according to PR branch.
+   * branched and removed automatically from `systemsmanagement:saltstack:github/salt` by OBS workflow.
+
+The same OBS structure will apply to all our OBS targets: `products`, `products:testing` and `products:next`.
+
+### Packaging artifacts
+
+All current extra "Sources" files in our RPM package, together with spec file and changelog file will go now to a `pkg/suse/` directory in `openSUSE/salt`:
+
+```
+pkg/suse/README.SUSE
```

Comment:
Not really. We won't be carrying patch files anymore in our new OBS packages, as any code change will automatically inside the source tarball by the `obs_scm` integration.

The exception to this would be EMBARGOED bugs, where we cannot proceed publicly via GitHub, so we will put a patch file manually in IBS than will be removed the bug is public and we push the changes to GitHub.

Reply:
A side question; currently, we have a lot of commits that are not in upstream, due to various reasons. With `salt-packaging`, we can easily check which patches are in upstream, and which patches are not. With removing the patch workflow, we're losing this insight.

Do you perceive this as a problem? Are we still sticking with the "fork & cherrypick commits" development style, or are we moving towards rebasing more frequently?

Reply:
This is a good point! We should probably want to have a way to easily identify what is upstreamed and what is not.

I'll elaborate this a bit on the RFC text. Thanks for the note :+1:

Reply:
thank you @m-czernek , that was exactly my concern when I made the comment. Commits that are not yet upstrem, and currently are maintained on salt-packaging project. 
We we start merging commit to our salt project that are not merge upstream yet, it can make it harder to integrate upstream version and know what is merge already or not.
@meaksh thank you for look into this topic.

Reply:
I've added a section in to RFC to clarify this.